pkg/server/service/loadbalancer/wrr/wrr.go

```diff
@@ -162,12 +174,17 @@ func (b *Balancer) nextServer() (*namedHandler, error) {
 		handler.deadline += 1 / handler.weight
 
 		heap.Push(b, handler)
-		if _, ok := b.status[handler.name]; ok {
-			if _, ok := b.fenced[handler.name]; !ok {
-				// do not select a fenced handler.
-				break
-			}
+		if _, down := b.status[handler.name]; !down {
+			continue
+		}
+		if _, isFenced := b.fenced[handler.name]; isFenced {
+			continue
+		}
+		if handler.passiveHealthChecker != nil && !handler.passiveHealthChecker.AllowRequest() {
```

Comment:
I think there might be an issue: if there's one server up but it has a passive health check that disallows requests, we wait until `failTimeout` expires. 
Is that the intended behavior? Should we instead return an `errNoAvailableServer` in this case?